app/controllers/api/projects_controller.rb

```diff
@@ -17,7 +17,12 @@ def index
     end
 
     def show
-      render :show, formats: [:json]
+      if !@project.school_id.nil? && @project.lesson_id.nil?
+        @project_with_user = @project.with_user(@current_user)
+        render :show_with_user, formats: [:json]
+      else
+        render :show, formats: [:json]
+      end
```

Comment:
💭 you might want to use the same show view and conditionally add the user field in the view so
```suggestion
        @project = @project.with_user(@current_user)
      end
      render :show, formats: [:json]
```